File: java/src/main/behavioral/mediator/ConcreteColleagueA.java
package behavioral.mediator;

// ConcreteColleagueA: A specific Colleague that communicates via the Mediator
public class ConcreteColleagueA extends Colleague {
    public ConcreteColleagueA(Mediator mediator) {
        super(mediator);
    }

    public void send(String message) {
        System.out.println("ConcreteColleagueA: Sending message -> " + message);
        mediator.sendMessage(message, this);
    }

    @Override
    public void receiveMessage(String message) {
        System.out.println("ConcreteColleagueA received message: " + message);
    }
}

File: python/src/behavioral/mediator/concrete_colleague_a.py
from .colleague import Colleague


class ConcreteColleagueA(Colleague):
    """
    Colleague A communicates through the mediator.
    """

    def do_action_a(self) -> str:
        if self._mediator:
            return self._mediator.notify(self, "A_ACTION")
        return "No mediator assigned."
